### ISSUE TITLE
Multiple listeners defined in `load_balancer`

### ISSUE THREAD

**Issue by drpebcak** [chef/chef-provisioning-aws#81]
```
load_balancer_options :availability_zones => ['us-west-1c'],
                          :subnet => 'subnet-52fb453a',
                          :listeners => [{
                            :port => 80,
                            :protocol => :http,
                            :instance_port => 80,
                            :instance_protocol => :http
                            },
                            {
                            :port => 443,
                            :protocol => :https,
                            :instance_port => 443,
                            :instance_protocol => :https,
                            :instance_certificate => 'EV-Cert-2015'
                            }]
```

Gives me `NameError: undefined local variable or method 'listener' for #<Chef::Provisioning::AWSDriver::Driver:0x007f854e60c888>`

**Comment by drpebcak** [comment]
Ok, so looks like you get that 'listener' error whenever the elb already exists.

Got this error when creating a new elb: `NoMethodError: private method 'select' called for nil:NilClass`
(It appears that the elb was configured regardless of this error)

And then this one when running the same thing again: `NameError: undefined local variable or method 'listener' for #<Chef::Provisioning::AWSDriver::Driver:0x007fa0271d7c10>`

**Comment by christinedraper** [comment]
I've seen the 'select' error - it happens when you create an ELB and don't include the 'machines' attribute.

I haven't managed to reproduce the 'listener' problem. Main difference between what I'm trying and your example above is my second listener is tcp because I dont have an SSL certificate. Guess it could be related to that.

**Comment by drpebcak** [comment]
Ah I did comment out the machine attribute while testing this - my nodes are Windows and chef-provisioning-aws has not yet implemented winrm. My nodes always fail the chef run!

**Comment by christinedraper** [comment]
I get a key error on 'instance_certificate'. From the code it looks like the key is 'server_certificate'. Don't know if that's relevant to your error

**Comment by drpebcak** [comment]
Yeah I found that too. It gets to the point where it creates the elb with the proper cert, and then it fails. Im going to test it with a *nix instance that will converge properly and see what happens.

**Comment by lynchc** [comment]
I just ran into both of these problems as well. They break down into their own issues:
```
NoMethodError: private method 'select' called for nil:NilClass
``` happens if you try to create a load balancer with no machines attached, as was mentioned. This is due to provisioning/aws_driver/driver.rb ln 171 where machine_spec is nil. This is easily fixed with a quick check, after which, it will complete properly.

The second error:
```
NameError: undefined local variable or method `listener' for <Chef::Provisioning::AWSDriver::Driver:0x00000002f24c78>
```
will happen whether the first run succeeds or not. This is due to provisioning/aws_driver/driver.rb ln 123. which, i'm guessing, is some left-over code that should have been removed. The code right below will do a comparison on desired vs. actual port. so I'm just commenting out ln. 123. 

I will submit a PR now with these fixes in place
